core/templates/pages/learner-dashboard-page/learner-dashboard-icons.component.ts

```diff
@@ -171,12 +174,37 @@ export class LearnerDashboardIconsComponent implements OnInit {
     }
   }
 
+  // This function will open a modal to remove an exploration
+  // from the 'Play Later' list in the Library Page.
   removeFromLearnerPlaylist(
       activityId: string, activityTitle: string, activityType: string): void {
-    this.learnerDashboardActivityBackendApiService
-      .removeFromLearnerPlaylistModal(
-        activityId, activityTitle, activityType,
-        this.learnerDashboardActivityIds);
+    // This following logic of showing a modal for confirmation previously
+    // resided in learnerDashboardActivityBackendApiService. However, in
+    // issue #14225, we noticed some errors with dynamic component creation.
+    // The componentFactoryResolver in the service was from AppModule and not
+    // the page specific module. This makes sense as the we provide all
+    // services in root (As specified by the providedIn: 'root'). So the
+    // injector used is the root injector and not the page module injector.
+    // The entry components specified in page module won't be available when
+    // we use the root injector.
+    // TODO(14290): Find a better way to refactor code that opens modals
+    // into new services that use the page specific injector rather than
+    // the root injector.
```

Comment:
This is not related to the migration right? It is a fully angular page causing the issue (trying to understand the scope). 

If yes for above, can this race condition happen on other modals too? Should we figure out how to handle those too? Is this a problem only when we call ngbModal within a service which might be getting provided in root? Would it not cause issues if we only trigger ngbModal on a component? If so that can be a viable pattern we introduce...

Reply:
I don't think it is a race condition. And I also think it is not the case for any other modal as of now. We are planning to do a full audit for all modals. As for the remaining modals we are migrating/ new modals that might be introduced we will keep the creation logic inside the component.